autotx/AutoTx.py

```diff
@@ -58,11 +70,30 @@ def __init__(
             silent=not config.verbose
         )
         self.agents = agents
+        self.log_costs = config.log_costs if config.log_costs is not None else False
```

Comment:
we could just make `log_costs: bool = False` in `Config` and pass it in the constructor if we want to use it, making things more readable

Reply:
Same issue as above with max_rounds

Reply:
If I make it not optional then it MUST be provided, which messes with the use in cli.